### PR TITLE
Add downlevel flag for storage buffers in vertex shaders

### PR DESCRIPTION
**Pull request by ElArtista**
**Description**
This will enable compatibility on systems that report support for storage buffers on fragment shaders but not on  vertex shaders.
This includes all Radeon Terrascale architecture GPUs.

### CODE REVIEW

File: wgpu-core/src/device/mod.rs [hunk 1]
```diff
@@ -1104,10 +1104,22 @@ impl<A: HalApi> Device<A> {
...
+            if entry.visibility.contains(wgt::ShaderStages::VERTEX) {
+                if writable_storage == WritableStorage::Yes {
+                    required_features |= wgt::Features::VERTEX_WRITABLE_STORAGE;
+                }
+                if let Bt::Buffer {
+                    ty: wgt::BufferBindingType::Storage { .. },
+                    ..
+                } = entry.ty
+                {
+                    self.require_downlevel_flags(wgt::DownlevelFlags::VERTEX_STORAGE)
```
Comment (cwfitzgerald) [inline wgpu-core/src/device/mod.rs:1116]: Could we do something similar as the required_features, enumerate them then require them once at the bottom.

**Review by cwfitzgerald — CHANGES_REQUESTED** [review]
One issue, nut otherwise lgtm

File: wgpu-core/src/device/mod.rs (hunk 1 above)
Comment (ElArtista) [reply wgpu-core/src/device/mod.rs]: Done as requested!

**Review by cwfitzgerald — APPROVED** [review]
Thanks!

**Pull request merged**